test/unit/loan/ExitQueuePSM.t.sol

```diff
@@ -0,0 +1,564 @@
+
+        // labels
+        vm.label(address(core), "core");
+        vm.label(address(profitManager), "profitManager");
+        vm.label(address(token), "token");
+        vm.label(address(credit), "credit");
+        vm.label(address(guild), "guild");
+        vm.label(address(psm), "exitqueuepsm");
+        vm.label(address(this), "test");
+    }
+
+    // constructor params & public state getters
+    function testInitialState() public {
+        assertEq(address(psm.core()), address(core));
+        assertEq(psm.profitManager(), address(profitManager));
+        assertEq(psm.credit(), address(credit));
+        assertEq(psm.pegToken(), address(token));
+        assertEq(psm.decimalCorrection(), 1e12);
+
+        // specifics for ExitQueuePSM
+        assertEq(psm.MIN_AMOUNT(), MIN_AMOUNT);
+        assertEq(psm.MIN_WITHDRAW_DELAY(), MIN_DELAY);
+        (uint256 totalCredit, uint256 totalCostPegToken) = psm
+            .getTotalCreditInQueue();
+        assertEq(totalCredit, 0);
+        assertEq(totalCostPegToken, 0);
+    }
+
+    // enforce that:
+    // - mint moves a number of tokens equal to getMintAmountOut() i/o
+    // - redeem moves a number of token equal to getRedeemAmountOut() i/o
+    // - rounding errors are within 1 wei for a mint/redeem round-trip
+    // same test as the simplePSM exactly, checks that when the queue is empty
+    // the contracts performs just like the SimplePSM
+    function testMintRedeem(uint256 input) public {
+        uint256 mintIn = (input % 1e15) + 1; // [1, 1_000_000_000e6]
+
+        // update creditMultiplier
+        // this ensures that some fuzz entries will result in uneven divisions
+        // and will create a min/redeem round-trip error of 1 wei.
+        credit.mint(address(this), 100e18);
+        assertEq(profitManager.creditMultiplier(), 1e18);
+        profitManager.notifyPnL(address(this), -int256((input % 90e18) + 1), 0); // [0-90%] loss
+        assertLt(profitManager.creditMultiplier(), 1.0e18 + 1);
+        assertGt(profitManager.creditMultiplier(), 0.1e18 - 1);
+        credit.burn(100e18);
+
+        assertEq(token.balanceOf(address(this)), 0);
+        assertEq(credit.balanceOf(address(this)), 0);
+        assertEq(token.balanceOf(address(psm)), 0);
+        assertEq(credit.balanceOf(address(psm)), 0);
+        assertEq(psm.pegTokenBalance(), 0);
+
+        // mint
+        token.mint(address(this), mintIn);
+        token.approve(address(psm), mintIn);
+        psm.mint(address(this), mintIn);
+
+        uint256 mintOut = psm.getMintAmountOut(mintIn);
+        assertEq(token.balanceOf(address(this)), 0);
+        assertEq(credit.balanceOf(address(this)), mintOut);
+        assertEq(token.balanceOf(address(psm)), mintIn);
+        assertEq(credit.balanceOf(address(psm)), 0);
+        assertEq(psm.pegTokenBalance(), mintIn);
+
+        // redeem
+        credit.approve(address(psm), mintOut);
+        psm.redeem(address(this), mintOut);
```

Comment:
Should redeem be locked is exit queue not empty?

I think it's a non issue because if the queue is not empty then it means there is nothing to redeem (or a user would have entered the exit queue while there was some assets to be redeemed?), if there is nothing to redeem then users cannot redeem (duh!)

But i can add an override and revert when users try to redeem when the queue is not empty (forcing them to enter the exit queue in that case)